Assets/MRTK/Core/Providers/InputAnimation/InputRecordingBuffer.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Container used to efficiently store a sequence of input animation keyframes while recording
+    /// </summary>
+    public class InputRecordingBuffer : IEnumerable<InputRecordingBuffer.Keyframe>
+    {
+        /// <summary>
+        /// The input state for a single frame
+        /// </summary>
+        public class Keyframe
+        {
+            public float Time;
```

Comment:
Can these all be public properties instead of fields? To align with the [MRTK coding guidelines around encapsulation](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#encapsulation).